### ISSUE TITLE
pick from compat adds extra 'undefined'

### ISSUE THREAD

**Issue by Xiot** [toss/es-toolkit#698]
It was mentioned that this issue was fixed last week, however I can still reproduce it in `1.24.0`
https://github.com/toss/es-toolkit/issues/236#issuecomment-2384876951

```typescript
import * as _ from 'lodash'
import * as es from 'es-toolkit'
import * as compat from 'es-toolkit/compat'

const o = { a: 1, b: 2 }
console.log('lodash    ', _.pick(o, ['a', 'c']))
console.log('es-toolkit', es.pick(o, ['a', 'c']))
console.log('compat    ', compat.pick(o, ['a', 'c']))
```

This produces
```
lodash     { a: 1 }
es-toolkit { a: 1 }
compat     { a: 1, c: undefined }
```

**Comment by raon0211** [comment]
Seems that there's also an issue in our compatibility version. Let me check this.

**Comment by raon0211** [comment]
This was resolved in es-toolkit v1.25.0. Please refer to [this commit](https://github.com/toss/es-toolkit/commit/28f0eaeb15de8bef0aea9844173ac59f3070d76c) for details. Thanks for finding an issue in our library!
